llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -6359,6 +6359,66 @@ bool SIInstrInfo::isOperandLegal(const MachineInstr &MI, unsigned OpIdx,
   return isImmOperandLegal(MI, OpIdx, *MO);
 }
 
+bool SIInstrInfo::isNeverCoissue(MachineInstr &MI) const {
```

Comment:
Is it possible to extract this information from the scheduler model without maintaining this list of instructions? i.e. there should be some execution resource you can check if the instruction uses